src/pf.lua

```diff
@@ -12,6 +12,10 @@ local ssa = require('pf.ssa')
 local backend = require('pf.backend')
 
 function compile_filter(filter_str, opts)
+   if opts.optimize == nil then
```

Comment:
I think we'd really be better with a declaration of what options there are and what defaults there are, so that users can know what's up.

e.g.

```lua
local parse_opts(opts, defaults)
  local ret = {}
  for k, v in pairs(opts) do
      if defaults[k] == nil then error('unrecognized option '..k) end
      ret[k] = v
  end
  for k, v in pairs(defaults) do
    if ret[k] == nil then ret[k] = v end
  end
  return ret
end
--- defaults list helps the reader, is a good place for documentation
local defaults { optimize=true, pcap_offline_filter=false, ...}
function compile_filter(filter_str, opts)
  local opts = parse_opts(opts, defaults)
  ....
end